### PR TITLE
Move Passport auth to API

### PR DESCRIPTION
**Pull request by sedubois**

### CODE REVIEW

**Comment by sedubois** [comment]
Needs merge of API for tests to pass

File: server/routes.js [hunk 1]
```diff
@@ -35,11 +34,12 @@ module.exports = (app) => {
    */
   app.use(robots(path.join(__dirname, '../frontend/dist/robots.txt')));
 
-  /**
-   * Authentication
-   */
-  app.get('/api/auth/:service(github)/:slug', aN.authenticateService);
-  app.get('/auth/:service(github)/callback/:slug', aN.authenticateServiceCallback);
+  // For some reason during OAuth, the redirect through piping automatically changes 
+  // the body, but not the window location -> instead redirect manually
```
Comment (sedubois) [inline server/routes.js]: @arnaudbenard any idea why this is needed and if there's a better fix?

**Comment by asood123** [comment]
code looks fine. will wait for tests to pass. then, i can rebase the `github-integration` branch on top of this

File: server/routes.js (hunk 1 above)
Comment (arnaudbenard) [inline server/routes.js]: I've never done redirects with piping, I found a SO question about it

http://stackoverflow.com/questions/13954094/redirect-express-js-piped-req-pipe-res-properly

Comment (sedubois) [inline server/routes.js]: Great that works, thanks for googling ;)

Comment (sedubois) [inline server/routes.js]: Fix: https://github.com/OpenCollective/opencollective-website/pull/253

Comment (arnaudbenard) [inline server/routes.js]: Awesome 💃

**Pull request merged**